board/app/models/thread.php

```diff
@@ -1,122 +1,180 @@
-    const MAX_TITLE_LENGTH = 30;
+const MIN_TITLE_LENGTH = 1;
+const MAX_TITLE_LENGTH = 30;
 
-//Thread Length Validation
+    /* 
+    * Thread Length Validation
+    */
     public $validation = array(
-        'title' => array(
-            'length' => array(
-                'validate_between', self::MIN_TITLE_LENGTH, self::MAX_TITLE_LENGTH
+    'title' => array(
+        'length' => array(
+            'validate_between', self::MIN_TITLE_LENGTH, self::MAX_TITLE_LENGTH
+            ),
+        ),
+
+    /* 
+    * Category Validation
+    */
+    'category' => array(
+        'length' => array(
+            'validate_between', self::MIN_TITLE_LENGTH, self::MAX_TITLE_LENGTH
             ),
         ),
     );
 
-    public function create(Comment $comment) 
+    public function create(Comment $comment)
     {
         $this->validate();
         $comment->validate();
+        
+        if ($this->hasError() || $comment->hasError()) {
+            throw new ValidationException('Invalid thread or comment');
+        }
+            
+        try {
+            $db = DB::conn();
+            $date_created = date("Y-m-d H:i:s");
+            $db->begin();
+            $params = array(  
+                'title' => $this->title, 
+                'created'=> $date_created, 
+                'user_id'=> $this->user_id,
+                'category' => $this->category
+                );
+            $db->insert('thread', $params); 
+            $this->id = $db->lastInsertId();
+            $comment->write($comment, $this->id); //<--write comment at the same time
+            $db->commit();
+        } catch (Exception $e) {
+            $db->rollback();
+        }
+    }
 
-            if ($this->hasError() || $comment->hasError())
-            {
-                throw new ValidationException('Invalid thread or comment');
-            }
+    public function isUserThread()     
+    {
+        return $this->user_id === $_SESSION['user_id'];
+    }
 
-        $date_created = date("Y-m-d H:i:s");
-        $params = array(
-            'title' => $this->title,
-            'created'=> $date_created
-             );
-
-    //Latest inserted ID
-            try 
-            {
-                $db = DB::conn();
-                $db->begin();
-                $db->insert('thread', $params);
-                $this->id = $db->lastInsertId();
-
-    //write first comment at the same time
-                $this->write($comment);
-                $db->commit();
-            } catch (Exception $e) {
-                $db->rollback();
-            }
+    public function deleteThread($user_id) 
+    {
+        try {
+            $db = DB::conn();
+            $db->begin();
+            $params = array(
+                $this->id,
+                $this->user_id
+            );
+            $db->query('DELETE FROM thread WHERE id = ? AND user_id = ?', $params);
+            $db->commit();
+            Comment::deleteAllComments($this->id);
```

Comment:
Since you want to deleteAll from the Comment model (avoid redundancy on the deleteAllComments as well), this line should be before the $db->commit(); :bow: